### PR TITLE
HBASE-26176 Correct regex in hbase-personality.sh

### PR DESCRIPTION
**Pull request by petersomogyi**
This is a rework of the original commit to enable
Hadoop 3 profile for the HBASE-25853 branch.

### CODE REVIEW

**Comment by petersomogyi** [comment]
This is a commit that we don't want to merge back to the base (branch-2) branch once HBASE-25853 is ready.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console in case of problems.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console in case of problems.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console in case of problems.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m  4s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests |  |
| uname | Linux 899d77e9d963 4.15.0-151-generic #157-Ubuntu SMP Fri Jul 9 23:07:57 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 50 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 11s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests |  |
| uname | Linux ec296ec8ce3d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 51 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 53s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  8s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  1s |  ASF License check generated no output?  |
|  |   |   3m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux c75ac2cee549 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 47 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/1/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
How about branch-2? This regex will only accept branch-2.x?

**Comment by petersomogyi** [comment]
> How about branch-2? This regex will only accept branch-2.x?

The regex will accept branch-2 and branch-2<anything> from now on. The previous regex was faulty and allowed `branch-1` too.

**Comment by Apache9** [comment]
> > How about branch-2? This regex will only accept branch-2.x?
> 
> The regex will accept `branch-2` and `branch-2<anything>` from now on. The previous regex was faulty and allowed `branch-1` too.

I'm not an expert of regex in bash but skimmed the script, we also have this

```
if [[ "${PATCH_BRANCH}" = branch-1* ]]; then
    yetus_info "Setting Hadoop 3 versions to test based on branch-1.x rules."
    hbase_hadoop3_versions=""
```

Is this also wrong?

**Comment by Apache9** [comment]
But also this

```
elif [[ "${PATCH_BRANCH}" = branch-2.* ]]; then
    yetus_info "Setting Hadoop 2 versions to test based on branch-2.3+ rules."
```

I'm a bit confused now...
What does the dot actually mean here? It means the dot itself or any char?

**Comment by petersomogyi** [comment]
I'll fix these regexes as well.

> What does the dot actually mean here? It means the dot itself or any char?

The dot means any character.

**Comment by petersomogyi** [comment]
Using `=` or `=~` is handled differently. The first uses pattern matching the latter uses regex.

```
bash-5.1$ export PATCH_BRANCH=branch-1; if [[ "${PATCH_BRANCH}" = branch-1* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-1.2; if [[ "${PATCH_BRANCH}" = branch-1* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-2; if [[ "${PATCH_BRANCH}" = branch-1* ]]; then echo "true"; fi
bash-5.1$ export PATCH_BRANCH=branch-; if [[ "${PATCH_BRANCH}" = branch-1* ]]; then echo "true"; fi

bash-5.1$ export PATCH_BRANCH=branch-1; if [[ "${PATCH_BRANCH}" =~ branch-1* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-1.2; if [[ "${PATCH_BRANCH}" =~ branch-1* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-2; if [[ "${PATCH_BRANCH}" =~ branch-1* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-; if [[ "${PATCH_BRANCH}" =~ branch-1* ]]; then echo "true"; fi
true

bash-5.1$ export PATCH_BRANCH=branch-1; if [[ "${PATCH_BRANCH}" =~ branch-1.* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-1.2; if [[ "${PATCH_BRANCH}" =~ branch-1.* ]]; then echo "true"; fi
true
bash-5.1$ export PATCH_BRANCH=branch-2; if [[ "${PATCH_BRANCH}" =~ branch-1.* ]]; then echo "true"; fi
bash-5.1$ export PATCH_BRANCH=branch-; if [[ "${PATCH_BRANCH}" =~ branch-1.* ]]; then echo "true"; fi
```
Based on this I can simplify the code. :)

**Comment by petersomogyi** [comment]
I've modified the PR for master. Please take a look at #3567. I'll update this PR for the feature branch once master is merged.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console in case of problems.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console in case of problems.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console in case of problems.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests |  |
| uname | Linux 226d7b6d6fea 4.15.0-151-generic #157-Ubuntu SMP Fri Jul 9 23:07:57 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 51 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 58s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests |  |
| uname | Linux 2e81acf280cd 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 52 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-25853 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  1s |  ASF License check generated no output?  |
|  |   |   2m 52s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3568 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux c63871027497 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-25853 / 4fbc4c29f2 |
| Max. process+thread count | 46 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3568/2/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/hbase-personality.sh [hunk 1]
```diff
@@ -464,7 +465,8 @@ function shadedjars_rebuild
   # If we have HADOOP_PROFILE specified and we're on branch-2.x, pass along
   # the hadoop.profile system property. Ensures that Hadoop2 and Hadoop3
   # logic is not both activated within Maven.
-  if [[ -n "${HADOOP_PROFILE}" ]] && [[ "${PATCH_BRANCH}" =~ branch-2* ]] ; then
+    if [[ -n "${HADOOP_PROFILE}" ]] \
```
Comment (taklwu) [inline dev-support/hbase-personality.sh:468]: nit: this lines seems to have two extra spaces before `if`

**Review by taklwu — APPROVED** [review]
since this is going to be a feature branch-only commit, I will still merge it without fixing the indentation

**Pull request merged**